ond.py

```diff
@@ -46,10 +46,53 @@ def get_config(self):
             },
         }
 
+    def _run_round(
+        self, algorithm, round_id: int, dataset_path: str,
+        algo_test_params: dict, algo_test_data: dict,
+        save_features: str = None, use_saved_features: bool = False,
+        use_feedback: bool = False
+    ):
+        with open(dataset_path, "r") as dataset:
+            dataset_ids = dataset.readlines()
+            image_ids = [image_id.strip() for image_id in dataset_ids]
+            algo_test_params["dataset_ids"].extend(image_ids)
+
+        if config["use_saved_features"]:
+            # TODO
+            pass
+        else:
+            (
+                algo_test_data["features_dict"],
```

Comment:
I would recommend using `update` here rather than assigning since you would need to aggregate features across multiple rounds to save them